guides/hyperlane-use-tia-for-gas.md

```diff
@@ -0,0 +1,446 @@
+# - Hyperlane agent-config  (default path: {cw-hyperlane-root}/context/stride-internal-1.config.json)
+yarn cw-hpl deploy -n stride-internal-1
+```
+
+#### Setup the relayer config:
+
+```bash
+echo '{
+  "db": "/etc/data/db",
+  "relayChains": "localwasmd,strideinternal1",
+  "allowLocalCheckpointSyncers": "true",
+  "gasPaymentEnforcement": [{ "type": "none" }],
+  "whitelist": [
+    {
+      "origindomain": [1063],
+      "destinationDomain": [1651]
+    },
+    {
+      "origindomain": [1651],
+      "destinationDomain": [1063]
+    }
+  ],
+  "chains": {
+    "localwasmd": {
+      "signer": {
+        "type": "cosmosKey",
+        "key": "0xf0517040b5669e2d93ffac3a3616187b14a19ad7a0657657e0f655d5eced9e31",
+        "prefix": "wasm"
+      }
+    },
+    "strideinternal1": {
+      "signer": {
+        "type": "cosmosKey",
+        "key": "0xf0517040b5669e2d93ffac3a3616187b14a19ad7a0657657e0f655d5eced9e31",
+        "prefix": "stride"
+      }
+    }
+  }
+}' > example/hyperlane/relayer.json
+```
+
+#### Setup the validator config on the Stride side:
+
+```bash
+echo '{
+  "db": "/etc/data/db",
+  "checkpointSyncer": {
+    "type": "localStorage",
+    "path": "/etc/validator/strideinternal1/checkpoint"
+  },
+  "originChainName": "strideinternal1",
+  "validator": {
+    "type": "hexKey",
+    "key": "0xf0517040b5669e2d93ffac3a3616187b14a19ad7a0657657e0f655d5eced9e31"
+  },
+  "chains": {
+    "strideinternal1": {
+      "signer": {
+        "type": "cosmosKey",
+        "key": "0xf0517040b5669e2d93ffac3a3616187b14a19ad7a0657657e0f655d5eced9e31",
+        "prefix": "stride"
+      }
+    }
+  }
+}' > example/hyperlane/validator.strideinternal1.json
+```
+
+#### Setup the validator config on the localwasmd rollup side:
+
+```bash
+echo '{
+  "db": "/etc/data/db",
+  "checkpointSyncer": {
+    "type": "localStorage",
+    "path": "/etc/validator/localwasmd/checkpoint"
+  },
+  "originChainName": "localwasmd",
+  "validator": {
+    "type": "hexKey",
+    "key": "0xf0517040b5669e2d93ffac3a3616187b14a19ad7a0657657e0f655d5eced9e31"
+  },
+  "chains": {
+    "localwasmd": {
+      "signer": {
+        "type": "cosmosKey",
+        "key": "0xf0517040b5669e2d93ffac3a3616187b14a19ad7a0657657e0f655d5eced9e31",
+        "prefix": "stride"
+      }
+    }
+  }
+}' > example/hyperlane/validator.localwasmd.json
+```
+
+#### Prepare the validators and relayer config:
+
+```bash
+# Create agent-config.docker.json by merging localwasmd.config.json and stride-internal-1.config.json
+jq -s '(.[0] | .chains.localwasmd.grpcUrl = .chains.localwasmd.grpcUrls[0].http) * (.[1] | .chains.strideinternal1.grpcUrl = .chains.strideinternal1.grpcUrls[0].http)' context/{localwasmd,stride-internal-1}.config.json > example/hyperlane/agent-config.docker.json
+```
+
+#### Update the docker compose file:
+
+```bash
+echo 'services:
+  relayer:
+    container_name: hpl-relayer
+    image: gcr.io/abacus-labs-dev/hyperlane-agent:3bb4d87-20240129-164519
```

Comment:
I believe the latest is `8a66544-20240530-111322`

Reply:
That's what they sent me on tg, will def try yours

Reply:
ah interesting. not sure which is more recent, but the one I sent they published last week